### PR TITLE
Trips example

### PR DESCRIPTION
**Pull request by menusal**
## Description

https://github.com/CartoDB/deck.gl-examples/assets/5357085/9cf3672a-9e44-4cbb-97ba-9ff21b7d6d45

Create an example for [this](https://codesandbox.io/s/carto-deck-gl-examples-rimo9x?from-embed=&initialpath=/advanced-examples/trips-layer.html&file=/advanced-examples/trips-layer.html)

You will need [this](https://github.com/visgl/deck.gl/blob/master/docs/api-reference/carto/overview.md#support-for-other-deckgl-layers) method from @felixpalmer. In the mean time, you can just run a query against SQL API for this

Shorcut: [sc-361053](https://app.shortcut.com/cartoteam/story/361053/trips-examples)

## CHECKLIST

- [x] Migrate old example
- [x] Use `query` from: [Support for other deck.gl layers](https://github.com/visgl/deck.gl/blob/master/docs/api-reference/carto/overview.md#support-for-other-deckgl-layers)

### CODE REVIEW

File: trips/.env [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# API Base URL (copy this from CARTO Workspace -> Developers)
+VITE_API_BASE_URL=https://gcp-us-east1.api.carto.com
+# Connection ID (app.carto.com -> Developers -> Manage API Access Tokens)
+VITE_API_CONNECTION_NAME=carto_dw
+# Set this variable if you're using an API Access Token (public application).
+# Go to app.carto.com -> Developers -> Manage API Access Tokens
+VITE_API_ACCESS_TOKEN=eyJhbGciOiJIUzI1NiJ9.eyJhIjoiYWNfbHFlM3p3Z3UiLCJqdGkiOiJkOTU4OWMyZiJ9.78MdzU2J6y-J6Far71_Mh7IQO9eYIZD9nECUiZJAVL4
```
Comment (alasarr) [inline trips/.env:7]: Missing end line

File: trips/index.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    <meta charset="UTF-8" />
+    <title>Deck.gl - Carto Trips example</title>
+  </head>
+  <body>
+    <div id="map"></div>
+    <canvas id="deck-canvas"></canvas>
+    <div id="story-card">
+      <p class="overline">✨👀 You're viewing</p>
+      <h2>Deck.gl - Carto Trips example</h2>
+      <p>Request data directly using the <b>CARTO SQL API</b> to integrate it with other deck.gl layers</p>
```
Comment (alasarr) [inline trips/index.html:13]: Adapt the following description

**Review by alasarr — CHANGES_REQUESTED** [review]
Besides the suggested code changes, I noticed that the example does not look as good as the trips layers at deck.gl. https://github.com/visgl/deck.gl/tree/8.9-release/examples/website/trips.

I'm aware it doesn't look like the original one, but it's the perfect time to fix it.

We need to add the buildings with an extrusion. Use carto-demo-data.demo_tables.manhattan_pluto_data and use the numfloors as the height of the building: https://clausa.app.carto.com/map/d966cc41-34ec-4754-b3bf-d6207ec1c737

File: trips/index.html (hunk 2 above)
Comment (menusal) [reply trips/index.html:13]: I changed it, but I'm unsure if it will be the final description.
Any suggestion?

File: trips/package.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "carto-deckgl-trips-example",
+  "version": "0.0.0",
+  "private": true,
+  "scripts": {
```
Comment (alasarr) [inline trips/package.json:5]: ALign with other examples: missing links and remove ocular-dev-tools

**Review by alasarr — CHANGES_REQUESTED** [review]
The visualization should look like https://deck.gl/examples/trips-layer (without the 3D buildings) but it doesn't.

File: trips/package.json (hunk 3 above)
Comment (menusal) [reply trips/package.json:5]: Done in this example, I'll check the rest, it's legacy

**Comment by menusal** [comment]
> carto-demo-data.demo_tables.manhattan_pluto_data

height: `numfloors` * 4

**Review by alasarr — CHANGES_REQUESTED** [review]
Still a different appearance:

This PR:
image

https://deck.gl/examples/trips-layer:
image

Buildings should be filtered and animation start with more lines

**Comment by menusal** [comment]
> Still a different appearance:
> 
> This PR: image
> 
> https://deck.gl/examples/trips-layer: image
> 
> Buildings should be filtered and animation start with more lines

@alasarr Sources are different, comes from https://codesandbox.io/p/sandbox/carto-deck-gl-examples-rimo9x?file=%2Fadvanced-examples%2Ftrips-layer.html [sc-361053](https://app.shortcut.com/cartoteam/story/361053/trips-examples)

## Buildings
In our example, we use `carto-demo-data.demo_tables.manhattan_pluto_data` and `numfloors` for building height. The Deck example uses [this source](https://raw.githubusercontent.com/visgl/deck.gl-data/master/examples/trips/buildings.json) with different heights and polygons.

## Trips
In our example, we use `cartobq.public_account.new_york_trips`.The deck example uses [this source](https://raw.githubusercontent.com/visgl/deck.gl-data/master/examples/trips/trips-v7.json)

I should try to adjust the initial time stamp in the query.

**Comment by alasarr** [comment]
You can filter by numFloors>20 to reduce the number of buildings in the viz

File: trips/index.ts [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+      map.jumpTo({center: [longitude, latitude], ...rest});
+    }
+  });
+
+  const data = await query({
+    ...cartoConfig,
+    sqlQuery: 'SELECT geom, vendor, TO_JSON_STRING(timestamps) AS timestamps FROM cartobq.public_account.new_york_trips'
+  });
+
+  const buildings = await query({
```
Comment (alasarr) [inline trips/index.ts:86]: The current payload is too big and the visualization has too many buildings to appreciate the lines of the trips (which is the important part of the viz)

Apply a stronger filter (>20 for example) and visualize it as a VectorTileLayer.

**Review by alasarr — CHANGES_REQUESTED** [review]

File: trips/index.ts (hunk 4 above)
Comment (menusal) [reply trips/index.ts]: Done

**Review by alasarr — APPROVED** [review]

**Pull request merged**